### PR TITLE
TIP-221: move catalog repository interfaces

### PR DESCRIPTION
**Pull request by jjanvier**
This PR moves catalog repository interfaces into the catalog component.

before: 30 coupling rules broken
after: 27 coupling rules broken

This PR solves a lot of repositories coupling problems and introduce new (little) ones (that will be resolved in another PR):
* 2 repos depends from Enrich
* 1 repo depends from UI
* 1 repo depends from Doctrine

```
| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | -
| Added Behats                      | -
| Changelog updated                 | Y
| Review and 2 GTM                  | 
| Micro Demo to the PO (Story only) | -
| Migration script                  | - 
| Tech Doc                          | -
```

### CODE REVIEW

File: src/Pim/Bundle/BaseConnectorBundle/Reader/Doctrine/ODMProductReader.php [hunk 1]
```diff
@@ -6,11 +6,10 @@
 use Akeneo\Component\Batch\Model\StepExecution;
 use Doctrine\ODM\MongoDB\DocumentManager;
 use Pim\Bundle\BaseConnectorBundle\Reader\ProductReaderInterface;
-use Pim\Bundle\BaseConnectorBundle\Validator\Constraints\Channel as ChannelConstraint;
```
Comment (skeleton) [inline src/Pim/Bundle/BaseConnectorBundle/Reader/Doctrine/ODMProductReader.php]: do not remove this line :)

File: src/Pim/Bundle/BaseConnectorBundle/Reader/Doctrine/ORMProductReader.php [hunk 2]
```diff
@@ -7,11 +7,10 @@
 use Doctrine\ORM\AbstractQuery;
 use Doctrine\ORM\EntityManager;
 use Pim\Bundle\BaseConnectorBundle\Reader\ProductReaderInterface;
-use Pim\Bundle\BaseConnectorBundle\Validator\Constraints\Channel as ChannelConstraint;
```
Comment (skeleton) [inline src/Pim/Bundle/BaseConnectorBundle/Reader/Doctrine/ORMProductReader.php]: same

**Comment by nidup** [comment]
GTM

**Comment by skeleton** [comment]
gtm once fixed

**Pull request merged**